machine/v1beta1/types_azureprovider.go

```diff
@@ -183,6 +183,19 @@ type OSDisk struct {
 	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// DiffDiskSettings describe ephemeral disk settings for the os disk.
+	// +optional
+	DiffDiskSettings DiffDiskSettings `json:"diffDiskSettings,omitempty"`
+	// CachingType specifies the caching requirements.
+	// +optional
+	CachingType string `json:"cachingType,omitempty"`
+}
+
+// DiffDiskSettings describe ephemeral disk settings for the os disk.
+type DiffDiskSettings struct {
+	// Option enables ephemeral OS when set to "Local"
```

Comment:
I think the linked doc suggests this is logically an `EphemeralStorageLocation` with options of "Local", "Remote", and "", where "" is no opinion, letting the platform choose a good default, which is subject to change over time.  Currently the default is "Remote'?

Also, great linking that doc, it's very helpful.

Reply:
I'm not sure what `EphemeralStorageLocation` is... Currently DiffDiskSettings support just `Option` which can be only `Local`. Other values are not supported. 
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.compute.models.diffdisksettings.option?view=azure-dotnet#Microsoft_Azure_Management_Compute_Models_DiffDiskSettings_Option

Reply:
from the linked page

> Ephemeral OS disks are created on the local virtual machine (VM) storage and not saved to the remote Azure Storage

This field controls `EphemeralStorageLocation`, which is more descriptive than `Option`.  I'm open to other potential names, but `Option` is insufficiently specific.   We produce a kube-like API to expose various configuration options in a manner we think is most consumeable by those knowledgeable in kubernetes.  If some work is required to translate that into the systems the platform interacts with on the backend, that is ok.